solver/src/settlement_submission/archer_settlement.rs

```diff
@@ -0,0 +1,377 @@
+                MINED_TX_PROPAGATE_TIME.as_secs()
+            );
+
+            loop {
+                if let Some(hash) = find_mined_transaction(&self.web3, &transactions).await {
+                    tracing::info!("found mined transaction {}", hash);
+                    return Ok(Some(hash));
+                }
+                if Instant::now() + MINED_TX_CHECK_INTERVAL > tx_to_propagate_deadline {
+                    break;
+                }
+                tokio::time::sleep(MINED_TX_CHECK_INTERVAL).await;
+            }
+        }
+
+        tracing::info!("did not find any mined transaction");
+        Ok(None)
+    }
+
+    async fn nonce(&self) -> Result<U256> {
+        self.web3
+            .eth()
+            .transaction_count(self.account.address(), None)
+            .await
+            .context("transaction_count")
+    }
+
+    /// Keep polling the account's nonce until it is different from initial_nonce returning the new
+    /// nonce.
+    async fn wait_for_nonce_to_change(&self, initial_nonce: U256) -> U256 {
+        const POLL_INTERVAL: Duration = Duration::from_secs(1);
+        loop {
+            match self.nonce().await {
+                Ok(nonce) if nonce != initial_nonce => return nonce,
+                Ok(_) => (),
+                Err(err) => tracing::error!("web3 error while getting nonce: {:?}", err),
+            }
+            tokio::time::sleep(POLL_INTERVAL).await;
+        }
+    }
+
+    async fn gas_price(&self, gas_limit: f64, time_limit: Duration) -> Result<f64> {
+        match self
+            .gas_price_estimator
+            .estimate_with_limits(gas_limit, time_limit)
+            .await
+        {
+            Ok(gas_price) if gas_price <= self.gas_price_cap => Ok(gas_price),
+            Ok(gas_price) => Err(anyhow!(
+                "gas station gas price {} is larger than cap {}",
+                gas_price,
+                self.gas_price_cap
+            )),
+            Err(err) => Err(err),
+        }
+    }
+
+    /// Keep submitting the settlement transaction to the archer network as gas price changes.
+    ///
+    /// Returns when simulation of the transaction fails. This likely happens if the settlement
+    /// becomes invalid due to changing prices or the account's nonce changes.
+    ///
+    /// Potential transaction hashes are communicated back through a shared vector.
+    async fn submit_with_increasing_gas_prices_until_simulation_fails(
+        &self,
+        deadline: SystemTime,
+        target_confirm_time: Duration,
+        nonce: U256,
+        settlement: Settlement,
+        gas_estimate: U256,
+        transactions: &mut Vec<H256>,
+    ) -> MethodError {
+        const UPDATE_INTERVAL: Duration = Duration::from_secs(10);
+
+        // The amount of extra gas it costs to include the payment to block.coinbase interaction in
+        // an existing settlement.
+        let gas_estimate = gas_estimate + U256::from(22859);
```

Comment:
I don't remember exactly but I think I got this value by comparing an empty settlement to a settlement that only has this interaction. The latter cost this much more gas. Maybe I made a mistake or maybe there is some more overhead for non empty settlements that I attributed to this interaction when I shouldn't have.

Reply:
I see, I thought it was the standalone transaction price since it was mentioned before and was similar. The comparison approach makes a lot of sense, here is what I see:

| Interactions in settlement | Gas cost | Difference without tip | 
| ------------- | ------------- | ------------- |
| [none](https://dashboard.tenderly.co/tx/rinkeby/0xc28ccc35fbee495248adaa23fba4fdedb0c240bf6e0634a4c1ff532b81564101/gas-usage) | 45844  | - |
| [tip](https://dashboard.tenderly.co/tx/rinkeby/0x885bff5eba9b80e51e1df9d6441d909761f1cc897f233b144ad04188d6c3885a/gas-usage)  | 64226  | 18382|
| [other](https://dashboard.tenderly.co/tx/rinkeby/0x9c073dfc29495e3e4e7ce57dbc35a1a7a998e1cd1397b9feee3c045f87856166/gas-usage)  | 56075  | - |
| [other + tip](https://dashboard.tenderly.co/tx/rinkeby/0xbdee3247de2c9ea8b51e0d34dee5ce13e75b3e001840c58f058ec90eba101861/gas-usage)  | 74421 | 18346 |

So my estimate was about 3k gas off, not sure where this extra gas comes from.

Reply:
I will change it to `18346` then. Thanks for measuring.